rfc/0000-configuration-and-shared-files.md

```diff
@@ -0,0 +1,296 @@
+
+- `{ident}`
+
+  To store configuration specific to a particular language implementation or tool. It is up to the implementation or tool to define the meaning of these entries, and each implementation or tool should only read entries under its own identifier.
+
+  Sub-tables may used to group configuration data within each `{ident}` section.
+
+  ```toml
+  [ruby]
+  bar = "baz"
+
+  [net.foo]
+  disabled = true
+
+  [pactflow]
+  enable_other_feature = false
+  ```
+
+#### Path Handling
+
+To ensure a uniform experience across the Pact ecosystem and operating systems, the following rules _must_ be followed when handling paths in the configuration file:
+
+- Paths _must_ be stored using the forward slash (`/`) as the path separator.
+  - This avoids the need to escape backslashes (`\`) in Windows paths.
+  - If a backslash is used, it is undefined behaviour. The implementation _may_ attempt to convert the backslashes to forward slashes, or may use the value as-is.
+- The tilde (`~`) character _must_ be expanded to the user's home directory if it is the first character in a path.
+  - For example, parsing `~/.local/share/pact` should result in `/home/user/.local/share/pact` on Unix-like systems.
+  - If the tilde appears elsewhere in the path, it is treated as a literal character.
+  - If someone wishes to use a literal tilde for a relative path, they _must_ use `./~` instead.
+- Relative paths are allowed, and _must_ be resolved relative to the configuration file.
+  - For example, if the configuration file is located at `/workdir/project/.pact.toml`, and a value is set to `.pact/data`, the resolved path should be `/workdir/project/.pact/data`.
+
+
+### Shared Files
+
+At present, there are only a few files which are shared across the Pact ecosystem. These are:
+
+- Plugins, currently stored in `~/.pact/plugins`
+- The Pact core library (i.e., the Pact FFI), though only in some implementations.
+
+As with the configuration files, a standard location is proposed for shared files which follows the operating system conventions:
+
+- If the `PACT_DATA_HOME` environment variable is set, this location _must_ be used.
+- Unix and Unix-like systems: `$XDG_DATA_HOME/pact/` (which defaults to `~/.local/share/pact/` if `XDG_DATA_HOME` is not set)
+- Windows: `%APPDATA%\pact\`
+- Fallback: `~/.pact/`
+
+> [!NOTE]
+>
+> macOS is considered a Unix-like system for the purposes of this RFC. Should an end-user prefer to use a different location (such as `~/Library/pact/`), they may do so by setting the `XDG_DATA_HOME` environment variable.
+
+If the operating system cannot be detected, the fallback location _should_ be used.
+
+If the operating system is detected and the fallback location is found, a warning _should_ be displayed to inform the user to relocate the directory. This exception to this is if the location of the shared files is explicitly set, in which case the warning _must not_ be displayed.
```

Comment:
These 2 sentences appear twice and I don't really understand them.

Can you explain them a bit more? Or an example for the warning message would be nice.

Reply:
Nevermind, after reading again, it's basically telling me to move my `~/.pact/plugins` into `~/.local/share/pact/plugins`